### ISSUE TITLE
Getting Error: apiRequestContext.get: connect ETIMEDOUT while executing api in playwright

### ISSUE THREAD

**Issue by pawanbrn** [microsoft/playwright#27017]
### System info
- Playwright Version: [v1.37.1]
- Operating System: [Windows 11]
- Browser: [Chromium]
- Other info:

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

async post(request: APIRequestContext, bearerToken: any, URL: string) {
    var Uri = `${URL}`;
    const response = await request.post(Uri, {
      headers: {
        Authorization: "Bearer " + bearerToken,
        "Content-Type": "application/json",
      },
      data: {
        paths: 'abc'
      },
      timeout: 0
    });
    const body = await response.text();
    return body;
  }

or

**Config file**

import type { PlaywrightTestConfig } from '@playwright/test';
import { devices } from '@playwright/test';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
const config: PlaywrightTestConfig = {
   testMatch: '/test.*spec.ts',
  /* Maximum time one test can run for. */
  timeout: 30 * 60 * 1000,
  globalSetup: "src/helper/globalSetup.ts",
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 5000
  },
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 5 : 5,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  //reporter: 'html',
  reporter: [['line'],['allure-playwright', {
    detail: true,
    outputFolder: 'allure-results',
    suiteTitle: false
   }],['junit', { outputFile: 'results.xml' }]],

  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
    actionTimeout: 0,
    trace: 'on-first-retry',
    screenshot : 'on',
    headless: false,
    channel: 'chrome',
    //viewport: { width: 1920, height: 1080 },
    
    viewport: null,
    launchOptions:{
      args: ["--start-maximized"],https://github.com/microsoft/playwright/issues/27017#top
    } 
  }
};

export default config;

**Expected**
APIRequestContext gets timeout with error Getting Error: apiRequestContext.get: connect ETIMEDOUT
This issue is intermittent and there is no server side issue

**Actual**

Should have allow to proceed without timeout.

**Note**
Initially Timeout was given close to 2 mins but did not resolve that and that is the reason have tried default timeout i.e.   timeout: 0

I can't share repo because its project related but its pretty much same as above api snippet just had to change uri and body

**Comment by dgozman** [comment]
We need more information to act on this issue. Could you please provide a repro that we can run locally?

**Comment by Amofe2** [comment]
I agree with @dgozman, you will need to share more information on this issue to be clear.
But also, pls confirm that {URL} holds a valid URL 

Are you trying to use string literals then it should be var Uri = `${URL}`

Confirm there is no server-side issue: That the server you are making request to is really responsive

You set  timeout: 0, which means no timeout. If there is an issue on the server or with the network, your code might hang indefinitely. Consider setting a reasonable timeout value to avoid indefinite waits.

Ensure that the Playwright context and page are set up correctly before this function is called. Playwright operates asynchronously, and if you're trying to make a request before the page is fully loaded or the context is ready, it can lead to timeouts.

Also your config file looks suspicious:

You might look at this example from https://playwright.dev/docs/api-testing
export default defineConfig({
  use: {
    // All requests we send go to this API endpoint.
    baseURL: 'https://api.github.com',
    extraHTTPHeaders: {
      // We set this header per GitHub guidelines.
      'Accept': 'application/vnd.github.v3+json',
      // Add authorization token to all requests.
      // Assuming personal access token available in the environment.
      'Authorization': `token ${process.env.API_TOKEN}`,
    },
  }
});

**Comment by pawanbrn** [comment]
I have updated details along with playwright.config.ts

**Comment by dgozman** [comment]
@pawanbrn Unfortunately, those details do not help. Since we cannot run your snippet due to missing `url`, I am not sure how to help you. Perhaps your API call does indeed timeout, so everything works as expected?

**Comment by dgozman** [comment]
Closing because we lack the information to act on this issue. If you can provide a full repro, please file a new issue by filling the "Bug Report" template, and link to this one.
